### PR TITLE
GitHub svg media library preview support

### PR DESCRIPTION
**Pull request by Jinksi**
**- Summary**

In the media library, SVG media loaded from https://raw.githubusercontent.com was unable to be displayed due to github's response content-type set to *text/plain* rather than *image/svg+xml*.

Appending a `sanitize=true` query to the url allows github to return the correct content-type.  

**- Test plan**

Example working SVG preview:  
screen shot 2017-12-21 at 3 32 27 pm

**- Description for the changelog**

Enable SVG preview in media library.

**- A picture of a cute animal (not mandatory but encouraged)**

emu

### CODE REVIEW

**Comment by verythorough** [comment]
Deploy preview ready!

Built with commit d1b4f892bb87f90ba53da975638d4df5fea685df

https://deploy-preview-954--netlify-cms-www.netlify.com

**Comment by verythorough** [comment]
Deploy preview ready!

Built with commit d1b4f892bb87f90ba53da975638d4df5fea685df

https://deploy-preview-954--cms-demo.netlify.com

**Comment by Jinksi** [comment]
Info on `sanitize=true` from [this stackoverflow answer](https://stackoverflow.com/a/16462143/7497911).

**Comment by erquhart** [comment]
This is awesome, thanks @Jinksi!

**Comment by tech4him1** [comment]
@erquhart Can you double-check this in Safari? According to [MDN](https://developer.mozilla.org/en-US/docs/Web/API/URL), `URL.searchParams` isn't supported in Safari or Edge. I'm guessing we would be able to polyfill it if necessary.

File: src/backends/github/implementation.js [hunk 1]
```diff
@@ -94,7 +94,11 @@ export default class GitHub {
     return this.api.listFiles(this.config.get('media_folder'))
       .then(files => files.filter(file => file.type === 'file'))
       .then(files => files.map(({ sha, name, size, download_url, path }) => {
-        return { id: sha, name, size, url: download_url, path };
+        const url = new URL(download_url);
+        if (url.pathname.match(/.svg$/)) {
+          url.searchParams.append('sanitize', true);
```
Comment (tech4him1) [inline src/backends/github/implementation.js]: MS Edge does not support `searchParams`, and it seems to break the entire media library if a single SVG is added.

Comment (tech4him1) [reply src/backends/github/implementation.js]: It looks like Edge will support this in v17: https://caniuse.com/#feat=urlsearchparams. We normally support two versions, though, so it's not practical to just wait until it is released.

Comment (tech4him1) [reply src/backends/github/implementation.js]: These should be equivalent:
```js
url.searchParams.append('sanitize', true);
```
```js
url.search += (url.search.slice(1) === '' ? '?' : '&') + 'sanitize=true';
```

Comment (tech4him1) [reply src/backends/github/implementation.js]: @Jinksi @erquhart What do you think  about using my example code above instead? ^^^^^^^^^

Comment (erquhart) [reply src/backends/github/implementation.js]: Good catch @tech4him1, agreed on the workaround.

Comment (Jinksi) [reply src/backends/github/implementation.js]: @tech4him1 @erquhart Updated https://github.com/netlify/netlify-cms/pull/954/commits/70c30905686c732f6b656063c4422a10f07bc2e8

**Comment by erquhart** [comment]
@tech4him1 this was updated, can you merge if you approve?

**Comment by tech4him1** [comment]
This looks fine for now, but there is a rendering issue in Microsoft Edge, so I've opened a separate issue for that: https://github.com/netlify/netlify-cms/issues/983.

**Pull request merged**